src/extensions/scratch3_text2speech/index.js

```diff
@@ -458,21 +550,29 @@ class Scratch3Text2SpeechBlocks {
         // Cast input to string
         let words = Cast.toString(args.WORDS);
 
+        if (words === '') return;
+
         const state = this._getState(util.target);
 
         const gender = this.VOICE_INFO[state.voiceId].gender;
         const playbackRate = this.VOICE_INFO[state.voiceId].playbackRate;
 
-        // @todo localize this?
         if (state.voiceId === KITTEN_ID) {
-            words = words.replace(/\w+/g, 'meow');
+            const meow = formatMessage({
+                id: 'text2speech.meow',
+                default: 'meow',
+                description: 'Meow call used for kitten voice.'
+            });
+            words = meow.repeat(Math.ceil(
+                words.replace(/[\x20-\x2f\x3a-\x40\x7b-\x7e]/g, '').length / 4
```

Comment:
This works for Japanese. Yet now this doesn't work for English!

What we want is:
- for Japanese (etc) to count by character
- for English (etc) to count by word

('meow' is a word, of course, and therefore must replace words)

It's also probably important to perform the division by 2 (like you did here -- though you divided by two twice, since you want two characters to become one meow, I think(?)) for non-Japanese characters (like emoji)

The solution for this is probably a little more complicated, therefore?

Reply:
Note: cats can no longer make dramatic pauses: "meow!meow" is pronounced differently from "meowmeow", but this change strips punctuation.

Haven't had a chance to review the other changes: but they look super good from a glance, nice job!

Reply:
On the other hand... splitting by syllables in English (etc) does makes some sense; maybe we should keep it like this. What are your thoughts?

Reply:
It's too hard to check if a language **always** uses whitespace to divide words.
And The kitten pronounces `antidisestablishmentarianism` and `hi` as `meow` - why is 28-letter word equal to 2-letter word?

Reply:
"an-ti-dis-es-ta-blish-men-tar-i-an-i-sm" 12 syllables vs "hi" 1 syllable is the more pressing question... unfortunately I don't think there's any good way to split words by syllable so splitting by words is the next best thing.